### PR TITLE
On timeline loaded

### PR DESCRIPTION
**Pull request by yotamberk**
Solves #3529

### CODE REVIEW

File: docs/timeline/index.html [hunk 1]
```diff
@@ -944,6 +944,14 @@ <h2 id="Configuration_Options">Configuration Options</h2>
     </tr>
 
     <tr>
+      <td>onInitialDrawComplete</td>
```
Comment (wimrijnders) [inline docs/timeline/index.html]: Small niggle on my part: The options are arranged alphabetically, this new entry and `onUpdate` are not.

File: examples/timeline/interaction/eventListeners.html [hunk 2]
```diff
@@ -34,7 +34,8 @@
 
     var container = document.getElementById('visualization');
     var options = {
-      editable: true
+      editable: true,
+      onInitialDrawComplete: function() { return logEvent('Timeline initial draw completed', {}) },
```
Comment (wimrijnders) [inline examples/timeline/interaction/eventListeners.html]: `return` not necessary. `logEvent` returns nothing.

File: lib/timeline/Core.js [hunk 3]
```diff
@@ -1003,7 +1003,6 @@ Core.prototype._redraw = function() {
   } else {
     this.redrawCount = 0;
   }
-  this.initialDrawDone = true;
```
Comment (wimrijnders) [inline lib/timeline/Core.js]: Doesn't this need to be set? It's the only place in the module where it goes to `true`.

OK, got it. It's set in `Timeline._activateOnInitialDrawDone()`. I find it a bit confusing that this is in a derived class.

File: lib/timeline/Timeline.js [hunk 4]
```diff
@@ -202,6 +202,16 @@ function Timeline (container, items, groups, options) {
 // Extend the functionality from Core
 Timeline.prototype = new Core();
 
+Timeline.prototype._activateOnInitialDrawDone = function() {
+  if (!this.initialDrawDone) {
+    this.initialDrawDone = true;
+    setTimeout(() => {
+      if (this.options.onInitialDrawComplete) {
```
Comment (wimrijnders) [inline lib/timeline/Timeline.js]: Why bother with `setTimeOut` if `onInitialDrawComplete` is not present? Better to move this condition out of `setTimeOut`.

File: lib/timeline/Timeline.js [hunk 5]
```diff
@@ -160,22 +161,21 @@ function Timeline (container, items, groups, options) {
...
           var range = me.getItemRange();
         }
 
         var start = me.options.start != undefined ? me.options.start : range.min;
         var end   = me.options.end   != undefined ? me.options.end   : range.max;
-        me.setWindow(start, end, {animation: false});
-      }
-      else {
-        me.fit({animation: false});
+        me.setWindow(start, end, {animation: false}, this._activateOnInitialDrawDone.bind(this));
```
Comment (wimrijnders) [inline lib/timeline/Timeline.js]: Strictly speaking, calling it here is a lie; `_redraw()` is done at the end of this method. `initialDrawDone` should be set to true *after* that call, same applies to the callback.

File: lib/timeline/Timeline.js [hunk 6]
```diff
@@ -411,21 +421,22 @@ Timeline.prototype.focus = function(id, options) {
  *                                    provided to specify duration and easing function.
  *                                    Default duration is 500 ms, and default easing
  *                                    function is 'easeInOutQuad'.
+ * @param {Function} callback
```
Comment (wimrijnders) [inline lib/timeline/Timeline.js]: `@param {function} [callback]`

- casing, callback optional

File: examples/timeline/interaction/eventListeners.html (hunk 2 above)
Comment (wimrijnders) [reply examples/timeline/interaction/eventListeners.html]: ;

File: lib/timeline/Core.js (hunk 3 above)
Comment (yotamberk) [reply lib/timeline/Core.js]: `Timeline._activateOnInitialDrawDone()` was incorrect. I removed it.

File: docs/timeline/index.html [hunk 7]
```diff
@@ -886,71 +886,71 @@ <h2 id="Configuration_Options">Configuration Options</h2>
...
       <td>Callback function triggered repeatedly when an item is being moved. See section <a href="#Editing_Items">Editing Items</a> for more information. Only applicable when both options <code>selectable</code> and <code>editable.updateTime</code> or <code>editable.updateGroup</code> are set <code><code>true</code></code>.
       </td>
     </tr>
-
+    
     <tr>
       <td>onRemove</td>
       <td>function</td>
       <td>none</td>
       <td>Callback function triggered when an item is about to be removed: when the user tapped the delete button on the top right of a selected item. See section <a href="#Editing_Items">Editing Items</a> for more information. Only applicable when both options <code>selectable</code> and <code>editable.remove</code> are set <code><code>true</code></code>.
```
Comment (wimrijnders) [inline docs/timeline/index.html]: Er, why the double `<code>` tags? It happens more often in the docs.

File: lib/timeline/Core.js [hunk 8]
```diff
@@ -111,6 +111,11 @@ Core.prototype._create = function (container) {
       this._redraw();
     }
   }.bind(this));
+  this.on('rangechanged', function () {
```
Comment (wimrijnders) [inline lib/timeline/Core.js]: Yes, this is a better place.

File: lib/timeline/Timeline.js [hunk 9]
```diff
@@ -173,9 +174,19 @@ function Timeline (container, items, groups, options) {
...
         var end   = me.options.end   != undefined ? me.options.end   : range.max;
-        me.setWindow(start, end, {animation: false}, this._activateOnInitialDrawDone.bind(this));
+        me.setWindow(start, end, {animation: false});
       } else {
-        me.fit({animation: false}, this._activateOnInitialDrawDone.bind(this));
+        me.fit({animation: false});
+      }
+    }
+
+    if (!me.initialDrawDone && me.initialRangeChangeDone) {
```
Comment (wimrijnders) [inline lib/timeline/Timeline.js]: Ah yes right. If I was just a bit clearer in my head I should have remarked on that. Make sense, not directly connected to the while-loop.

**Comment by yotamberk** [comment]
@wimrijnders seems like that `<code><code>` was in many places... thanks for noticing!

**Pull request merged**